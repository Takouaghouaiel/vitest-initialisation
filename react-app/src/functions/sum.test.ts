import { expect, test,expectTypeOf,vi,describe,it } from "vitest";
import { sum } from "./sum";


test("adds 1 + 2 to equal 3", () => {
  expect(sum(1, 2)).toEqual(3);
});

test("adds 1 + 2 + 3 to equal 4", () => {
  expect(sum(1, 2, 3)).toMatchSnapshot();
});

test("1 number", () => {
  expect(sum(1)).toBe(1);
});

test("no number", () => {
  expect(sum()).toBe(0);
});

test('my types work properly', () => {
  expectTypeOf(sum).toBeFunction();
})

test("try the mocking functions", () => {
  // about vi : vitest mocking api:
const mockSum = vi.fn();
mockSum(1, 2, 3);
expect(mockSum).toHaveBeenCalledWith(1, 2, 3);
expect(mockSum).toHaveBeenCalledTimes(1);
expect(vi.isMockFunction(mockSum)).toBe(true)
expect(mockSum.mock.calls[0]).toEqual([1, 2, 3])

})


describe('suite', () => {
  it.todo('unimplemented test')
})


it('snapshot', () => {
  const result = sum(1,4)
  expect(result).toMatchInlineSnapshot(`5`)
})