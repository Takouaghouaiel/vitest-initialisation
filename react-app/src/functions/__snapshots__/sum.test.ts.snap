// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`adds 1 + 2 + 3 to equal 4 1`] = `6`;
